library/MedEx/API.php

```diff
@@ -0,0 +1,2970 @@
+                            }  
+                               
+                            if ($logged_in) {
+                                if  (!empty($logged_in['products']['ordered'])) {
+                                    foreach ($logged_in['products']['ordered'] as $ordered) {
+                                        echo $ordered['menu'];
+                                    }
+                                }   
+                            }
+
+                            ?>
+                        </ul>
+                    </div><!-- /.navbar-collapse -->
+                </div>
+            </nav>
+        </div>
+        <?php
+        if ($GLOBALS['medex_enable'] == '1') {
+            $error=$this->MedEx->getLastError();
+            if (!empty($error['ip'])) {
+                ?>
+                <div class="alert alert-danger" style="width:50%;margin:30px auto 5px;font-size:0.9em;text-align:center;">    
+                <?php
+                echo $error['ip'];
+                ?>
+                </div>
+                <?php
+            }
+        }
+    }
+    public function preferences($prefs = '')
+    {
+        global $logged_in;
+        if (empty($prefs)) {
+            $prefs = sqlFetchArray(sqlStatement("SELECT * FROM medex_prefs"));
+        }
+        ?>
+        <div class="row">
+            <div class="col-sm-12 text-center">
+                <div class="showRecalls" id="show_recalls">
+                    <div class="title">MedEx <?php echo xlt('Preferences'); ?></div>
+                    <div name="div_response" id="div_response" class="form-inline"><br />
+                    </div>
+                    <form action="#" name="save_prefs" id="save_prefs">
+                            <div class="row">
+                                <input type="hidden" name="go" id="go" value="Preferences">
+                                <div class="col-sm-5 div-center col-sm-offset-1" id="daform2">
+                                    <div class="divTable2">
+                                        <div class="divTableBody prefs">
+                                            <div class="divTableRow">
+                                                <div class="divTableCell divTableHeading">MedEx <?php echo xlt('Username'); ?></div>
+                                                <div class="divTableCell indent20">
+                                                    <?php echo $prefs['ME_username']; ?>
+                                                </div>
+                                            </div>
+                                            <div class="divTableRow">
+                                                <div class="divTableCell divTableHeading"><?php echo xlt('General'); ?></div>
+                                                <div class="divTableCell indent20">
+                                                    <input type="checkbox" class="update" 
```

Comment:
TODO
-Make this PSR4 compatible and autoload (phpstorm should make this pretty easy for me to do)
-Minimal escaping and translation work
-Datepicker